src/CostModelDynamicLevelFactory.sol

```diff
@@ -29,9 +29,10 @@ contract CostModelDynamicLevelFactory is BaseModelFactory {
     uint256 costFactorAtZeroUtilization_,
     uint256 costFactorAtFullUtilization_,
     uint256 costFactorInOptimalZone_,
-    uint256 optimalZoneRate_
+    uint256 optimalZoneRate_,
+    bytes32 baseSalt_
   ) external returns (CostModelDynamicLevel model_) {
-    model_ = new CostModelDynamicLevel({
+    model_ = new CostModelDynamicLevel{salt: baseSalt_}({
```

Comment:
we need to manually set the salt so we can compute the address beforehand (solc uses a random salt if not manually specified like this)

Reply:
also ci fails because we need to update the PAT or something - I need to look into how to do that. will do that separately